packages/console/src/ds-components/TextInput/NumericInput.tsx

```diff
@@ -0,0 +1,89 @@
+import classNames from 'classnames';
+import { type ComponentProps } from 'react';
+
+import CaretDown from '@/assets/icons/caret-down.svg';
+import CaretUp from '@/assets/icons/caret-up.svg';
+import { onKeyDownHandler } from '@/utils/a11y';
+
+import * as styles from './NumericInput.module.scss';
+import TextInput from './index';
+
+type ButtonProps = {
+  className?: string;
+  onTrigger?: (
+    event: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>
+  ) => void;
+  children: React.ReactNode;
+  isDisabled?: boolean;
+};
+
+function Button({ className, onTrigger, children, isDisabled }: ButtonProps) {
+  return (
+    <div
+      role="button"
+      tabIndex={0}
+      className={classNames(styles.button, isDisabled && styles.disabled, className)}
```

Comment:
Should we use the <button> tag directly for such use case?

Reply:
no it's a `<div>`. i've tried `<button>` but met some strange styling issues

Reply:
ok. cc: @charIeszhao there were some places in AC where we have to use div to represent a button element by tweaking lots of a11y code. Maybe we could have a style-cleaned global button css class provided for such a case, so we could use button element directly.

Reply:
or we can figure out how to properly style a `<button>`. then no need for these tricky codes

Reply:
Maybe we can simply modify the use of `a11y.ts`, and make it return an object of props, including `role`, `tabIndex`, `onKeyDown` and `onClick`. Then we can spread and merge it into the target element. Thoughts?